### PR TITLE
Char types are UTF-16-encoded

### LINKED ISSUE #2173: .NET Framework reference, Char Struct, Char values and interop:  Error "UTF-8 encoded"
see: https://docs.microsoft.com/en-us/dotnet/api/system.char?view=netframework-4.7#remarks

section "Char values and interop"

existing text: "When a managed Char type, which is represented as a Unicode UTF-8 encoded code unit, is passed to unmanaged code, the interop marshaler converts the character set to ANSI."

Char type represents a UTF-16 code unit. 

new / correct text: "When a managed Char type, which is represented as a Unicode UTF-16 encoded code unit, is passed to unmanaged code, the interop marshaler converts the character set to ANSI."

### PR DESCRIPTION
**Pull request by rpetrusha**
# Char types are UTF-16-encoded

## Summary

The interop section of the Char structure documentation says that Char objects are UTF-8 encoded; they are not. 

Fixes #2173 

## Suggested Reviewers

@mairaw @b3bb

### CODE REVIEW

**Review by guardrex — APPROVED** [review]

**Review by mairaw — APPROVED** [review]
LGTM! Let's wait for green light to merge.

File: xml/System/Char.xml
Comment (svick) [inline xml/System/Char.xml]: Wouldn't it be clearer to say that it's converted to ANSI *by default*, considering that the two attributes can be used to change how it's converted? Something like:

> When a managed \<xref:System.Char> type, which is represented as a Unicode UTF-16 encoded code unit, is passed to unmanaged code, by default the interop marshaler converts the character set to ANSI. You can apply the \<xref:System.Runtime.InteropServices.DllImportAttribute> attribute to platform invoke declarations and the \<xref:System.Runtime.InteropServices.StructLayoutAttribute> attribute to a COM interop declaration to control which character set a marshaled \<xref:System.Char> type uses.

**Comment by rpetrusha** [comment]
Yes, it would be, @svick -- thanks for the suggestions. I'd already merged the PR before I saw your comment, so I'll open another PR with your suggestions. Again, thanks.

**Pull request merged**
